client/app/scripts/superdesk-dictionaries/dictionaries.js

```diff
@@ -102,74 +114,93 @@
             );
         };
 
-        fetchDictionaries();
+        $scope.fetchDictionaries();
     }
 
-    DictionaryConfigModalController.$inject = ['$scope', 'dictionaries', 'upload', 'gettext', 'notify', 'modal'];
-    function DictionaryConfigModalController ($scope, dictionaries, upload, gettext, notify, modal) {
-        function reset() {
-            $scope.dictionary = null;
-            $scope.origDictionary = null;
-            $scope.word = {};
-            $scope.file = null;
-            $scope.modalActive = false;
-        }
+    DictionaryEditController.$inject = ['$scope', 'dictionaries', 'upload', 'gettext', 'notify', 'modal'];
+    function DictionaryEditController ($scope, dictionaries, upload, gettext, notify, modal) {
 
-        var onSuccess = function(result) {
-            reset();
-            dictionaries.fetch(function(result) {
-                $scope.dictionaries = result;
-            });
+        function onSuccess(result) {
+            $scope.closeDictionary();
+            $scope.fetchDictionaries();
             notify.success(gettext('Dictionary saved succesfully'));
             return result;
-        };
+        }
 
-        var onError = function(response) {
+        function onError(response) {
             if (angular.isDefined(response.data._issues) &&
                     angular.isDefined(response.data._issues['validator exception'])) {
                 notify.error(gettext('Error: ' + response.data._issues['validator exception']));
             } else {
                 notify.error(gettext('Error. Dictionary not saved.'));
             }
-        };
+        }
 
-        //listen for the file selected event
+        // listen for the file selected event
         $scope.$on('fileSelected', function (event, args) {
             $scope.$apply(function() {
                 $scope.file = args.file;
             });
         });
 
-        $scope.canSave = function(form) {
-            var hasId = _.has($scope.dictionary, '_id') && $scope.dictionary._id !== null;
-            return form.$valid && (form.$dirty || $scope.file !== null) && (!hasId && $scope.file !== null || hasId);
-        };
-
         $scope.save = function() {
             if ($scope.file) {
-                dictionaries.upload($scope.dictionary, $scope.file, onSuccess, onError);
+                dictionaries.upload($scope.origDictionary, $scope.dictionary, $scope.file, onSuccess, onError);
             } else {
                 dictionaries.update($scope.origDictionary, $scope.dictionary, onSuccess, onError);
             }
         };
 
         $scope.cancel = function() {
-            reset();
+            $scope.closeDictionary();
+        };
+
+        $scope.addWord = function(word) {
+            $scope.dictionary.content = $scope.dictionary.content || [];
+            $scope.dictionary.content.push(word);
```

Comment:
yep but it handles adding and removing words together. and you can add many words and save it only once. so having extra endpoint seems like premature optimisation atm